### ISSUE TITLE
Too Few Arguments to SAWCore Backend?

### ISSUE THREAD

**Issue by WeeknightMVP** [GaloisInc/crucible#177]
As of early last week, when attempting to build `crucible-server` on CentOS 7 64-bit, I get a couple errors from `src/Lang/Crucible/Server/Verification/Override.hs`:

```
$ git pull
Already up-to-date.
$ ll stack.yaml
... stack.yaml -> stack-ghc-8.4.yaml
$ which hpb
~/.local/bin/hpb
$ ./scripts/build-sandbox.sh
$ stack setup
stack will use a sandboxed GHC it installed
For more information on paths, see 'stack path' and 'stack exec env'
To use this GHC and packages outside of a project, consider using:
stack ghc, stack ghci, stack runghc, or stack exec
$ stack build --extra-lib-dirs=/usr/local/lib
...
Completed 39 action(s).            
Log files have been written to: .../crucible/.stack-work/logs/

--  While building package crucible-server-0.1 using:
      .../crucible/crucible-server/.stack-work/dist/x86_64-linux/Cabal-2.2.0.1/setup/setup --builddir=.stack-work/dist/x86_64-linux/Cabal-2.2.0.1 build lib:crucible-server exe:crucible-server --ghc-options " -ddump-hi -ddump-to-file -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
    Logs have been written to: .../crucible/.stack-work/logs/crucible-server-0.1.log

    Configuring crucible-server-0.1...
    Running preBuild
    Preprocessing library for crucible-server-0.1..
    Building library for crucible-server-0.1..
    [14 of 15] Compiling Lang.Crucible.Server.Verification.Override ( src/Lang/Crucible/Server/Verification/Override.hs, .stack-work/dist/x86_64-linux/Cabal-2.2.0.1/build/Lang/Crucible/Server/Verification/Override.o )
    
    .../crucible/crucible-server/src/Lang/Crucible/Server/Verification/Override.hs:130:1: error:
        • The type synonym ‘SAW.SAWCoreBackend’ should have 3 arguments, but has been given 2
        • In the type synonym declaration for ‘SAWBack’
        |
    130 | type SAWBack n = SAW.SAWCoreBackend n (Flags FloatReal)
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    
    .../crucible/crucible-server/src/Lang/Crucible/Server/Verification/Override.hs:131:42: error:
        • Expecting one more argument to ‘SAWBack n’
          Expected a type, but ‘SAWBack n’ has kind ‘* -> *’
        • In the second argument of ‘OverrideSim’, namely ‘(SAWBack n)’
          In the type ‘OverrideSim p (SAWBack n) () r args ret a’
          In the type declaration for ‘N’
        |
    131 | type N p n r args ret a = OverrideSim p (SAWBack n) () r args ret a
        |                                          ^^^^^^^^^
```

**Comment by robdockins** [comment]
Oops, I have a fix for this that accidentally ended up on a feature branch.  I'll pull it into master.

**Comment by robdockins** [comment]
Should be fixed in c6705885c58003c090248f2a60febe7a166c8716
